packages/ember-auto-import/ts/analyzer-syntax.ts

```diff
@@ -0,0 +1,249 @@
+import { ReadStream } from 'fs';
+
+export interface LiteralImportSyntax {
+  isDynamic: boolean;
+  specifier: string;
+}
+
+export interface TemplateImportSyntax {
+  isDynamic: boolean;
+  // these are the string parts of the template literal. The first one always
+  // comes before the first expression.
+  cookedQuasis: string[];
+  // for each of the expressions in between the cookedQuasis, this is an
+  // optional hint for what to name the expression that goes there. It's
+  // optional because in general there may not be an obvious name, but in
+  // practice there often is, and we can aid debuggability by using names that
+  // match the original code.
+  expressionNameHints: (string | null)[];
+}
+
+export type ImportSyntax = LiteralImportSyntax | TemplateImportSyntax;
+
+const marker = 'eaimeta';
```

Comment:
thoughts:

* Do we need a version in the marker? So that upgrades to the format are friendly?
* Do we need a more unique marker? Or do you feel `eaimeta` is enough? my thought was: something like `|ember-auto-import@<meta-version>|`

Reply:
Versioning is a good question. 

A single copy of ember-auto-import does both the reading and writing, so that is not an issue. But since the babel cache is persistent, you could find things in the cache that were written by a different version. I think the best way to address that is to make sure our babel plugin config itself changes if the protocol changes. We can include the protocol version in the babel plugin config and that should take care of it, because AFAIK changing the babel config should invalidate the babel persistent cache.

Regarding uniqueness: because we prepend at the very beginning of each module and our reading side consumes the first copy found, we only need to resist collisions with anything that gets prepended in front of us. But since that includes the AMD wrapper and that includes everybody's module names, it's definitely possible to put stuff there that would collide.

I'll tackle both of these by picking a weird marker and by making the marker an explicitly piece of the babel config.